src/config/settings.ts

```diff
@@ -153,24 +153,31 @@ export function getUpdatableNode(cfg: Settings, fqFieldAccessor: string): [Recor
 }
 
 export function updateFromCommandLine(cfg: Settings, args: string[]): Settings {
-  let i = 0;
   const lim = args.length;
+  let processingExternalArguments = true;
 
-  // Not using a for-loop here because `i` gets incremented in the loop to handle cases of `--option value`
-  // ... in the belief that we don't usually expect the body of for-loops to manipulate the loop index.
-  while (i < lim) {
+  // As long as processingExternalArguments is true, ignore anything we don't recognize.
+  // Once we see something that's "ours", set processingExternalArguments to false.
+  for (let i = 0; i < lim; i++) {
```

Comment:
This still feels weird because of the `i += 1` in the `default:` case; we should at least put a warning in the comment, maybe?